File: 2088-minimum-time-to-type-word-using-special-typewriter/minimum-time-to-type-word-using-special-typewriter.py
class Solution:
    def minTimeToType(self, word: str) -> int:
        ans = 0
        pointer = "a"
        for char in word:
            clockwise_move = abs(ord(char) - ord(pointer))
            counter_clockwise_move = 26 - clockwise_move
            ans += min(clockwise_move, counter_clockwise_move) + 1
            pointer = char
        return ans

File: 2088-minimum-time-to-type-word-using-special-typewriter/minimum-time-to-type-word-using-special-typewriter.cpp
class Solution {
public:
    int minTimeToType(string word) {
        int ans = 0;
        int pointer = 'a';
        for (int i = 0; i < word.length(); i++) {
            int clockwise_move = abs(word[i] - pointer);
            int counter_clockwise_move = 26 - clockwise_move;
            ans += min(clockwise_move, counter_clockwise_move) + 1;
            pointer = word[i];
        }
        return ans;
    }
};